src/main/java/org/apache/ofbiz/ofbiztutorial/usermgr/events/UserEvents.java

```diff
@@ -0,0 +1,130 @@
+        String postalCode = request.getParameter("postalCode");
+        String countryGeoId = request.getParameter("countryGeoId");
+        String countryCode = request.getParameter("countryCode");
+        String areaCode = request.getParameter("areaCode");
+        String contactNumber = request.getParameter("contactNumber");
+        String emailAddress = request.getParameter("emailAddress");
+
+        Map<String, Object> errorMap=new HashMap<>();
+        List<String> errors=new ArrayList<>();
+        // Perform Validations
+        if(UtilValidate.isEmpty(userLoginId)){
+            errors.add("User Login Id cannot be empty!!");
+        }
+        if(UtilValidate.isEmpty(currentPassword)){
+            errors.add("Password cannot be left empty!!");
+        }
+        if(UtilValidate.isEmpty(currentPasswordVerify) && !currentPassword.equals(currentPasswordVerify)){
+            errors.add("Failed to verify password: Confirm password doesn't match the password!!");
+        }
+        if(UtilValidate.isEmpty(firstName)){
+            errors.add("First Name cannot be empty!!");
+        }
+        if(UtilValidate.isEmpty(lastName)){
+            errors.add("Last Name cannot be empty!!");
+        }
+//        if(UtilValidate.isEmpty(gender)){
+//            return ServiceUtil.returnError("Gender cannot be empty!!").toString();
+//        }
+//        if (!UtilValidate.isDate(birthDate)) {
+//            return ServiceUtil.returnError("Incorrect Date Format in Birth Date!!!" + birthDate).toString();
+//        }
+        if (!UtilValidate.isDate(birthDate)) {
+            errors.add("Incorrect Date Format in Birth Date!!!" + birthDate);
+        }
+        if(UtilValidate.isEmpty(address1)){
+            errors.add("Address Line 1 cannot be empty!!");
+        }
+        if(UtilValidate.isEmpty(city)){
+            errors.add("City cannot be empty!!");
+        }
+        if(UtilValidate.isEmpty(postalCode)){
+            errors.add("Postal code cannot be empty!!");
+        }
+        if(UtilValidate.isEmpty(countryGeoId)){
+            errors.add("Country is required field!!");
+        }
+        if(UtilValidate.isEmpty(countryCode)){
+            errors.add("Country Code of the phone number is required field!!");
+        }
+        if (UtilValidate.isEmpty(contactNumber)){
+            errors.add("Invalid contact number!!!");
+        }
+        if(!UtilValidate.isEmail(emailAddress)){
+            errors.add("Enter valid email address");
+        }
+
+        if(!errors.isEmpty()){
+            return ServiceUtil.returnError(errors).toString();
+        }
+        try {
+            Map<String, Object> serviceParams = new HashMap<>();
+            serviceParams.put("userLogin", userLogin);
+            serviceParams.put("userLoginId", userLoginId);
+            serviceParams.put("currentPassword", currentPassword);
+            serviceParams.put("currentPasswordVerify", currentPasswordVerify);
```

Comment:
please use makeValidContext method here

Reply:
To use the makeValidContext method first I have to create a context map and put all these values in the context map. Then pass that context in the makeValidContext method to get valid context for the service.